contrib/swagger.yaml

```diff
@@ -0,0 +1,439 @@
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+        - name: snippet
+          in: body
+          schema:
+            $ref: '#/definitions/Snippet'
+          required: true
+          description: |
+              A snippet object, which will be used for update. Can have all or
+              only some of the fields set. Only set fields will be used for
+              update. If new ``content`` is provided, a new changeset of the
+              snippet will be created and set as the current one.
+      responses:
+        201:
+          description: |
+            The snippet has been successfully updated. The latest state of the
+            snippet is returned.
+          schema:
+            $ref: '#/definitions/Snippet'
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Bad request.
+        401:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Unauthorized. User tries to update a snippet without authorizing first.
+        403:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Forbidden. User tries to update a snippet by another author or anonymous.
+        404:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Not found. Snippet with the given ID value does not exist.
+    delete:
+      summary: Delete the specific code snippet given an id value.
+      description: Delete the specific code snippet given an id value.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+      responses:
+        200:
+          description: The snippet has been successfully deleted.
+        401:
+          description: |
+            Unauthorized. Only authorized authors are allowed to delete code snippets.
+          schema:
+            $ref: '#/definitions/Error'
+        403:
+          description: |
+            Forbidden. Authors are only allowed to delete snippets they own.
+          schema:
+            $ref: '#/definitions/Error'
+        404:
+          description: The snippet not found.
+          schema:
+            $ref: '#/definitions/Error'
+
+  /snippets/{id}/changesets:
+    get:
+      summary: Retrieve all changesets of the snippet.
+      description: Retrieve all changesets of the snippet.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+      responses:
```

Comment:
I think `401`, `403` and `404` are also applied. Or you want to hide them and use `400` in all cases?